File: chaos-core-bukkit/src/main/java/de/chaosolymp/core/bukkit/command/GarbageCollectorCommand.java
package de.chaosolymp.core.bukkit.command;

import de.chaosolymp.core.bukkit.BukkitPlugin;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

public final class GarbageCollectorCommand implements CommandExecutor {

    private final BukkitPlugin plugin;

    public GarbageCollectorCommand(final BukkitPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if(sender.hasPermission("core.gc")) {
            System.gc();
            this.plugin.getLogger().info(String.format("%s has started the garbage collector.", sender.getName()));
            // TODO: send garbage collector done message
        } else {
            // TODO: send no permission message
        }
        return true;
    }
}

File: src/main/kotlin/de/chaosolymp/core/command/GarbageCollectorCommand.kt
package de.chaosolymp.core.command

import de.chaosolymp.core.CorePlugin
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class GarbageCollectorCommand(private val plugin: CorePlugin) : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if(sender.hasPermission("core.gc")) {
            System.gc()
            if(sender is Player) this.plugin.logger.info("${sender.name} started the garbage collector") else this.plugin.logger.info("Running garbage collector.")
            // TODO: Garbage collector message
        } else {
            // TODO: No permission message
        }
        return true
    }

}
